src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -187,6 +197,9 @@ public function loadClass($class)
                     if ($parent && isset(self::$deprecated[$parent]) && strncmp($ns, $parent, $len)) {
                         @trigger_error(sprintf('The %s class extends %s that is deprecated %s', $name, $parent, self::$deprecated[$parent]), E_USER_DEPRECATED);
                     }
+                    if ($parent && isset(self::$final[$parent]) && strncmp($ns, $parent, $len)) {
+                        @trigger_error(sprintf('Extending the class %s in %s is deprecated and may break in the next major release.', $parent, $name), E_USER_DEPRECATED);
```

Comment:
@greg0ire we could do as for `@deprecated` tags and prepend the triggered message with the annotation description. I'm not sure that's worth it too so let's leave it as is for now :)

@stof if people mock a class having a `@final` tag, a deprecation will be triggered; I don't think that's acceptable if we want to allow it. If we want to keep some `@final` annotation virtual, then I think we should add a special case when we want to allow mocks (we could add something special in the annotation description?).